tardis/io/util.py

```diff
@@ -12,6 +13,34 @@
 import logging
 logger = logging.getLogger(__name__)
 
+float_pattern = r"""    # A numeric string consists of:
+    (?P<sign>[-+])?              # an optional sign, followed by either...
+
+    (
+
+        (?=\d|\.\d)              # ...a number (with at least one digit)
+
+        (?P<int>\d*)             # having a (possibly empty) integer part
+
+        (\.(?P<frac>\d*))?       # followed by an optional fractional part
+
+        (E(?P<exp>[-+]?\d+))?    # followed by an optional exponent, or...
```

Comment:
Oh right, but I think we should use `e` as afaik most ppl use the lowercase letter.
To be more explicit I'd drop `re.IGNORECASE` and modify the regex slightly.